File: SWEA/[1234] 비밀번호/kw601/main.java
import java.util.*;
import java.io.*;

class Solution
{
	public static void main(String args[]) throws Exception
	{
		int T = 10;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		for(int test_case = 1; test_case <= T; test_case++)
		{
            StringTokenizer st = new StringTokenizer(br.readLine());
            int length = Integer.parseInt(st.nextToken());
            String password = st.nextToken();  

            Stack<Character> stack = new Stack<>();
            for(char c : password.toCharArray()){
                if (!stack.isEmpty() && stack.peek() == c){
                    stack.pop();
                }
                else{
                    stack.push(c);
                }
            }

            System.out.print("#" + test_case + " ");
            StringBuilder sb = new StringBuilder();
            for (char c : stack) {
                sb.append(c);
            }
            System.out.println(sb.toString());
		}
	}
}

File: SWEA/[1234] 비밀번호/kw601/main.py
T = 10
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    # 입력 받고
    length, password = map(str, input().split())
    # 스택 만들고
    stack = []

    for s in password:
        # 빈 스택이 아니고 peek == s이면
        if stack and stack[-1] == s:
            stack.pop()
        else:
            stack.append(s)

    result = "".join(stack)
    print(f"#{test_case} {result}")
